### PR TITLE
432 automate access expiration

### PR DESCRIPTION
**Pull request by Fancien**
Solves issue #432 by running a method every day at midnight which terminates the Access objects which have an expiration date lower than the current day.

settings_local.template.py has been modified to automatically create the Periodic Task.

### CODE REVIEW

File: core/tasks.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from datetime import date
+from celery import shared_task
+from core.models.access import Access
+
+
+@shared_task
+def check_accesses_expiration():
```
Comment (neoflex) [inline core/tasks.py:7]: would be good to add docstring here

File: core/models/access.py [hunk 2]
```diff
@@ -38,6 +38,33 @@ class Meta:
...
+        to terminated
+
+        @param upper_date: Date threshold.
+        """
+        accesses_to_expire = cls.objects.filter(
+            grant_expires_on__lt=upper_date
+        ).exclude(
+            status=StatusChoices.terminated,
+        ).all()
+        for access in accesses_to_expire:
```
Comment (neoflex) [inline core/models/access.py:54]: would be good to add some logging messages

File: core/tasks.py (hunk 1 above)
Comment (Fancien) [reply core/tasks.py:7]: Done.

File: core/models/access.py (hunk 2 above)
Comment (Fancien) [reply core/models/access.py:59]: Done.
I added one debug log in the loop to mention accesses that are expired and to show the comparison between the expiration date and the given parameter.

I added another one at the end to say that the change was done successfully.

**Review by neoflex — APPROVED** [review]

**Pull request merged**